src/envoy/utils/filter_state_utils.h

```diff
@@ -35,6 +38,14 @@ absl::string_view getStringFilterState(
     const Envoy::StreamInfo::FilterState& filter_state,
     absl::string_view data_name);
 
+void setErrorFilterState(Envoy::StreamInfo::FilterState& filter_state,
+                         const google::rpc::Status& status);
+
+bool hasErrorFilterState(Envoy::StreamInfo::FilterState& filter_state);
```

Comment:
we should be able to pass in const & since we are not modifying it.  same for the next function